src/layers/domain/fhir/r4/cpm_model.py

```diff
@@ -2,31 +2,44 @@
 from uuid import UUID
 
 from domain.core.validation import ENTITY_NAME_REGEX, ODS_CODE_REGEX
+from domain.ods import ODS_API_BASE
 from pydantic import BaseModel as _BaseModel
 from pydantic import Extra, Field
 
+SYSTEM = "connecting-party-manager"
+
 
 class BaseModel(_BaseModel, extra=Extra.forbid):
     pass
 
 
+def ConstStrField(value):
+    """Field that can only take the given value, and defaults to it"""
+    return Field(default=value, regex=rf"^{value}$")
+
+
 class Identifier(BaseModel):
-    id: str = Field(regex=ODS_CODE_REGEX)
-    value: str = Field(min_length=1)
+    system: str = ConstStrField(SYSTEM)
+    value: UUID
+
+    def dict(self, *args, **kwargs):
+        """Additionally serializes UUID to string"""
+        _dict = super().dict(*args, **kwargs)
+        (_, _dict["value"]) = self.value.urn.split("urn:uuid:")
+        return _dict
+
+
+class OdsIdentifier(BaseModel):
+    system: str = ConstStrField(ODS_API_BASE)
+    value: str = Field(regex=ODS_CODE_REGEX)
 
 
 class Reference(BaseModel):
-    identifier: Identifier
+    identifier: OdsIdentifier
```

Comment:
Does this tie us into only having a Reference that is an OdsIdentifier?

Reply:
nah, we can create a different reference class for references to other things - I guess it should probably be called `OdsReference` since it does a regex on the identifier. There is also the option to use generics:

```python
    class Reference(Generic, BaseModel)[T]:
        identifier: T
    
    ...

    class Organization(BaseModel):
        ...
        partOf: Reference[OdsIdentifier]
```

if that's easier to read? bit more extensible

Reply:
d-done